CMSgov/cmcs-eregulations
# EREGCSC-1184 - Resources Sidebar

Resolves [EREGCSC-1184](https://jiraent.cms.gov/browse/EREGCSC-1184)

**Description**
When viewing a Part, this story adds another way to view Subpart or Section resources: a sidebar. 

[See Figma design](https://www.figma.com/file/CgNr2dlFenI9b49NWGpkgU/eRegs-Multi-View-Pattern?node-id=665%3A11951) for more information.

- Visiting a part will default to the drawer view
  - https://d3nslflb3ukady.cloudfront.net/42/431 would show the drawer
- A third URL parameter after Title and Part may be added to manually set drawer or sidebar
  - https://d3nslflb3ukady.cloudfront.net/42/431/drawer would show the drawer
  - https://d3nslflb3ukady.cloudfront.net/42/431/sidebar would show the sidebar

**This pull request changes.**

- Adds a URL parameter after title and part to manually set drawer or sidebar resources views
- Adds resources sidebar to Part view per the [Figma design](https://www.figma.com/file/CgNr2dlFenI9b49NWGpkgU/eRegs-Multi-View-Pattern?node-id=665%3A11951)
- Re-uses existing Supplemental Content component to display supplemental content.  Should match look and feel of supplemental content on the Pilot site.
- Sidebar should be sticky and should scroll independent of the part view
- Adds ability to expand or collapse all supplemental content collapsable sections
- Adds non-functioning Search Resources text input in sidebar
- Adds non-functioning "View All Resources" button in sidebar

**Steps to manually verify this change**

1. Visit any section in the prototype.  The default view should be that the resources drawer opens when you click "View Section Resources."  Example:  https://d3nslflb3ukady.cloudfront.net/42/431
2. Add `/sidebar` to the end of the section URL.  You should see the Resources sidebar on the right side of the page with an empty state.  Clicking "View Section Resources" will remove the empty state, trigger the loading spinner, and display supplemental content for that subpart or section if content exists. Example: https://d3nslflb3ukady.cloudfront.net/42/431/sidebar
3. Add `/drawer` to the end of the section URL. This should explicitly set the resources view to the Drawer concept.  Example: https://d3nslflb3ukady.cloudfront.net/42/431/drawer
